iroh-metrics/src/macros.rs

```diff
@@ -115,18 +116,79 @@ macro_rules! make_metric_recorders {
                         )+
                     }
                 }
+
+                pub fn run(self, rx: std::sync::mpsc::Receiver<$crate::core::MMsg>) {
+                    tokio::task::spawn_blocking(move || {
```

Comment:
`spawn_blocking` is for inside of async fns, should probably use a tokio channel if you need to wait on them inside of async fns

Reply:
Ok, so this is the part that's messed up and some next lever rust-fu might be able to help. 

Since everything in metrics is basically a singleton, `Lazy` forces me to have everything be `Sync`. Thus I have to rely on `SyncSender` from the `std`.

Alongside that If I run multiple of these collectors, they seem to starve the scheduler completely, because they probably just alternate awaiting for a message that won't come just yet. Moving it to a blocking spawn ensures it's separate from the rest and can run on it's own accord.